tests/processing/strategies/test_streaming.py

```diff
@@ -405,6 +408,48 @@ def test_parallel_transform_worker_apply() -> None:
     smm.shutdown()
 
 
+def fail_bad_messages(message: Message[KafkaPayload]) -> KafkaPayload:
+    if message.payload.key is None:
+        raise InvalidMessage(str(message), "No Key")
+    return message.payload
+
+
+def test_parallel_transform_worker_bad_messages() -> None:
+    smm = SharedMemoryManager()
+    smm.start()
+    input_block = smm.SharedMemory(128)
+    output_block = smm.SharedMemory(128)
+    input_batch = MessageBatch[Any](input_block)
+
+    # every other message has a key
+    messages = [
+        Message(
+            Partition(Topic("test"), 0),
+            i,
+            KafkaPayload(None if i % 2 == 0 else b"key", b"\x00", []),
+            datetime.now(),
+        )
+        for i in range(10)
+    ]
+
+    for message in messages:
+        input_batch.append(message)
+
+    assert len(input_batch) == 10
+
+    index, output_batch, bad_messages = parallel_transform_worker_apply(
+        fail_bad_messages,
+        input_batch,
+        output_block,
+    )
+    # all 10 messages processed
+    assert index == 10
+    # 5 were bad, 5 were good
+    assert len(bad_messages) == 5
+    assert len(output_batch) == 5
```

Comment:
Not sure if the length of output batch has other logical implications, but out of 10 messages, 5 were processed successfully and 5 were raised together to DLQ.